assets/javascripts/discourse/components/chat-composer-uploads.js

```diff
@@ -0,0 +1,172 @@
+      // then again maybe not, because we may not want all processors
+      // for chat...
+      this.uploadPreProcessors.push({
+        pluginClass: UppyMediaOptimization,
+        optionsResolverFn: ({ isMobileDevice }) => {
+          return {
+            optimizeFn: (data, opts) =>
+              this.mediaOptimizationWorker.optimizeImage(data, opts),
+            runParallel: !isMobileDevice,
+          };
+        },
+      });
+    }
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    this._bindUploadTarget();
+
+    this.set(
+      "fullPage",
+      document.body.classList.contains("has-full-page-chat")
+    );
+
+    this.appEvents.on(
+      `${this.eventPrefix}:upload-success`,
+      this,
+      "_insertUpload"
+    );
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    this.set("_uploads", this.uploads || []);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+
+    this.set("uploadPreProcessors", null);
+    this.set("uploadProcessorActions", null);
+    this.set("uploadMarkdownResolvers", null);
+
+    this.appEvents.off(
+      `${this.eventPrefix}:upload-success`,
+      this,
+      "_insertUpload"
+    );
+  },
+
+  handleUploadState() {
+    this.onUploadsChanged?.(this._uploads, {
+      inProgressUploads: this.inProgressUploads,
+      isProcessingUpload: this.isProcessingUpload,
+    });
+  },
+
+  @discourseComputed("_uploads.[]", "inProgressUploads.[]")
+  showUploadsContainer() {
+    return this._uploads?.length > 0 || this.inProgressUploads?.length > 0;
+  },
+
+  @discourseComputed("fullPage")
+  fileUploadElementId(fullPage) {
+    return fullPage ? "chat-full-page-uploader" : "chat-widget-uploader";
+  },
+
+  @discourseComputed("fullPage")
+  mobileFileUploaderId(fullPage) {
+    return fullPage
+      ? "chat-full-page-mobile-uploader"
+      : "chat-widget-mobile-uploader";
+  },
+
+  @discourseComputed()
+  acceptedFormats() {
+    const extensions = authorizedExtensions(
+      this.currentUser.staff,
+      this.siteSettings
+    );
+
+    return extensions.map((ext) => `.${ext}`).join();
+  },
+  @discourseComputed()
+  acceptsAllFormats() {
+    return authorizesAllExtensions(this.currentUser.staff, this.siteSettings);
+  },
+
+  @action
+  cancelUploading(upload) {
+    this.appEvents.trigger(`${this.eventPrefix}:cancel-upload`, {
+      fileId: upload.id,
+    });
+    this.handleUploadState();
+  },
+
+  @action
+  removeUpload(upload) {
+    this._uploads.removeObject(upload);
+    this.handleUploadState();
+  },
+
+  _insertUpload(_, upload) {
+    this._uploads.pushObject(upload);
+    this.handleUploadState();
+  },
+
+  _findMatchingUploadHandler() {
+    return;
+  },
+
+  _uploadDropTargetOptions() {
+    const target = document.querySelector(".chat-live-pane");
+    if (!target) {
+      return this._super();
+    }
+
+    return { target };
+  },
+
+  _cursorIsOnEmptyLine() {
```

Comment:
This is another thing used in core's `_uploadPlaceholder` function that we don't care about...I have some thoughts forming now, will type them up in the final review box.

Reply:
Leaving this here but nevermind it, see the main PR comment.